pkg/client/command_line.go

```diff
@@ -15,62 +16,215 @@ func AddArmadaApiConnectionCommandlineArgs(rootCmd *cobra.Command) {
 	viper.BindPFlag("armadaUrl", rootCmd.PersistentFlags().Lookup("armadaUrl"))
 }
 
-func LoadCommandlineArgsFromConfigFile(cfgFile string) {
+// LoadCommandlineArgsFromConfigFile load config from at exePath/armadactl-defaults.yaml, where
+// exePath is the path to the armadactl executable, the provided cfgFile, or, if it cfgFile="",
+// $HOME/.armadactl
+func LoadCommandlineArgsFromConfigFile(cfgFile string) error {
+
+	// read config at exePath/armadactl-defaults.yaml, where exePath is the path to the armadactl
+	// executable (if it exists)
 	exePath, err := os.Executable()
 	if err != nil {
+		return fmt.Errorf("[LoadCommandlineArgsFromConfigFile] error finding executable path: %s", err)
+	}
 
-		log.Errorf("Can't find executable path because %s\n", err)
-		os.Exit(1)
-	} else {
-		exeDir := filepath.Dir(exePath)
-		viper.SetConfigFile(exeDir + "/armadactl-defaults.yaml")
-		err := viper.ReadInConfig()
-		if err != nil {
-			switch err.(type) {
-			case viper.ConfigFileNotFoundError:
-			case *os.PathError:
-				// No default config is fine
-			default:
-				log.Errorf("Can't read config file %s because %s\n", viper.ConfigFileUsed(), err)
-				os.Exit(1)
-			}
+	exeDir := filepath.Dir(exePath)
+	configPath := filepath.Join(exeDir, "/armadactl-defaults.yaml")
+	viper.SetConfigFile(configPath)
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			// Config file not found; ignore
```

Comment:
I'd be tempted to log this.  It may save someone some time one day when the defaults file ahs been incorrectly deleted and armadactl doesn't work.

Reply:
This does mean if you aren't setting an `armadactl-defaults.yaml` file. We'd log an error on every command.

Could we have a verbose logging option maybe?

Reply:
so it doesn't need to be an error-debug would be fine.  I'd assumed that `armada-defaults` would be installed as part of the client install (as we're assuming it's next to the exe) and therefore should generally be there?

Reply:
Its so package managers internal to companies can put a default there along with the installed exe/binary. 
 - So their users don't need to specify any armada config like api
 
 However if the user isn't using a package manager - or their company decides not to populate a armada-defaults, it is not expected to be present.

Reply:
Later, I want to add a `view` command to `armadactl`, which prints the merged config. This command could perhaps be made to print the source files used to produce the merged config (as suggested below), which might be the right tradeoff between clarity and verbosity.